### PR TITLE
[docs] [core] fixed reference to ray.remote in tasks index page

### LINKED ISSUE #42502: [Core] Incorrect link to ray.remote package reference
### Description

The link to `ray.remote` api doc after the first code cell points to the core API main page (https://docs.ray.io/en/latest/ray-core/api/index.html), while it should point to https://docs.ray.io/en/latest/ray-core/api/doc/ray.remote.html

### Link

https://docs.ray.io/en/latest/ray-core/tasks.html#ray-remote-functions

### PR DESCRIPTION
**Pull request by angelinalg**
Fixing broken link.

## Related issue number

Closes https://github.com/ray-project/ray/issues/42502

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/tasks.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ Ray enables arbitrary functions to be executed asynchronously on separate Python
             :start-after: __tasks_start__
             :end-before: __tasks_end__
 
-        See the `ray.remote package reference <package-ref.html>`__ page for specific documentation on how to use ``ray.remote``.
+        See the :ref: `ray.remote package reference <ray.remote>` page for specific documentation on how to use ``ray.remote``.
```
Comment (matthewdeng) [inline doc/source/ray-core/tasks.rst:17]:
```suggestion
        See the :ref:`ray.remote package reference <ray.remote>` page for specific documentation on how to use ``ray.remote``.
```

image

Comment (angelinalg) [reply doc/source/ray-core/tasks.rst:17]: Thank you for catching that. 🤦‍♀️

**Comment by angelinalg** [comment]
Screenshot 2024-01-29 at 12 01 27 PM

**Comment by angelinalg** [comment]
Screenshot 2024-01-29 at 2 35 10 PM

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**